controllers/medusa/medusabackupjob_controller.go

```diff
@@ -128,7 +128,12 @@ func (r *MedusaBackupJobReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		logger.Info("backup complete")
 
 		// The MedusaBackupJob is finished and we now need to create the MedusaBackup object.
-		if err := r.createMedusaBackup(ctx, backup, logger); err != nil {
+		backupSummary, err := r.getBackupSummary(ctx, backup, pods, logger)
+		if err != nil {
+			logger.Error(err, "Failed to get backup summary")
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
```

Comment:
suggestion: If err != nil, the controller-runtime will requeue in any case. Setting Requeue time when we know requeue is going to happen in any case is confusing. 

The RequeueAfter value is actually ignored, since the err != nil causes the controller-runtime to process that first and then do a RateLimited queueing:

https://github.com/kubernetes-sigs/controller-runtime/blob/8117577334339e46bc27a46dbf98a0112aef6bb4/pkg/internal/controller/controller.go#L322

Reply:
sounds good. I thought about that but tried to stay consistent with the rest of the code in this controller.
I'll create a refactoring ticket to make this change everywhere in the medusa controllers.